### PR TITLE
Cleanup API dependencies

### PR DESCRIPTION
**Pull request by simonbrandhof**
SONAR-6562 Remove sonar-squid 4.1 from API and runtime classpath

and continue SONAR-6517:
* Remove the XML libs provided by Java 7
* Remove picontainer dep from sonar-plugin-api
* Merge back sonar-deprecated into sonar-plugin-api in order to benefit from shaded and relocated dependencies

### CODE REVIEW

**Comment by SonarTech** [comment]
SonarQube analysis reported 42 new issues:
* 8 critical
* 8 major
* 2 minor
* 24 info

Details:
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/batch/AbstractSourceImporter.java#L37)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/batch/ResourceCreationLock.java#L33)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/batch/ResourceFilter.java#L38)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/batch/SquidUtils.java#L39)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/batch/SquidUtils.java#L47)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/batch/SquidUtils.java#L59)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/batch/SquidUtils.java#L29)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [org.sonar.api.charts.AbstractChart.COLORS should be package protected](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/charts/AbstractChart.java#L44)[](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3AMS_PKGPROTECT)
* [Make this array "private".](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/charts/AbstractChart.java#L44)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1873)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/charts/AbstractChart.java#L39)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/charts/Chart.java#L34)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Introduce a new variable instead of reusing the parameter "val".](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/charts/ChartParameters.java#L179)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1226)
* [Provide multiple methods instead of using "decode" to determine which action to take.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/charts/ChartParameters.java#L97)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2301)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/charts/ChartParameters.java#L39)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Use an "instanceof" comparison instead.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/checks/AnnotationCheckFactory.java#L118)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1872)
* [Use an "instanceof" comparison instead.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/checks/AnnotationCheckFactory.java#L121)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1872)
* [Use an "instanceof" comparison instead.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/checks/AnnotationCheckFactory.java#L130)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1872)
* [Use an "instanceof" comparison instead.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/checks/AnnotationCheckFactory.java#L127)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1872)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/checks/AnnotationCheckFactory.java#L44)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Use an "instanceof" comparison instead.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/checks/AnnotationCheckFactory.java#L115)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1872)
* [The Cyclomatic Complexity of this method "configureField" is 15 which is greater than 10 authorized.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/checks/AnnotationCheckFactory.java#L108)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
* [Use an "instanceof" comparison instead.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/checks/AnnotationCheckFactory.java#L124)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1872)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/checks/CheckFactory.java#L34)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/checks/NoSonarFilter.java#L37)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [org.sonar.api.resources.Java.SUFFIXES should be package protected](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/resources/Java.java#L56)[](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3AMS_PKGPROTECT)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/resources/Java.java#L34)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Make this array "private".](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/resources/Java.java#L56)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1873)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/resources/JavaFile.java#L35)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/resources/JavaPackage.java#L29)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Override this superclass' "equals" method.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/resources/Method.java#L26)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2160)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/resources/Method.java#L26)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/resources/ProjectUtils.java#L32)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/resources/ProjectUtils.java#L43)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Method 'Rule.setCardinality(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/rules/AnnotationRuleParser.java#L74)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/rules/AnnotationRuleParser.java#L46)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/rules/RuleRepository.java#L37)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Provide multiple methods instead of using "useKeyIfEmpty" to determine which action to take.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/rules/RuleRepository.java#L60)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2301)
* [The Cyclomatic Complexity of this method "processRule" is 15 which is greater than 10 authorized.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/rules/XMLRuleParser.java#L118)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/rules/XMLRuleParser.java#L56)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Method 'Rule.setCardinality(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/rules/XMLRuleParser.java#L153)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/web/AbstractDashboardWidget.java#L27)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/d5f3560b11a87b495bc92dc009ad812cc23944a7/sonar-plugin-api/src/main/java/org/sonar/api/web/Section.java#L27)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/test/java/org/sonar/api/rules/AnnotationRuleParserTest.java [hunk 1]
```diff
@@ -84,14 +85,14 @@ public void should_reject_invalid_property_types() {
...
   public void should_recognize_type() {
-    assertThat(AnnotationRuleParser.guessType(Integer.class)).isEqualTo(PropertyType.INTEGER);
-    assertThat(AnnotationRuleParser.guessType(int.class)).isEqualTo(PropertyType.INTEGER);
-    assertThat(AnnotationRuleParser.guessType(Float.class)).isEqualTo(PropertyType.FLOAT);
-    assertThat(AnnotationRuleParser.guessType(float.class)).isEqualTo(PropertyType.FLOAT);
-    assertThat(AnnotationRuleParser.guessType(Boolean.class)).isEqualTo(PropertyType.BOOLEAN);
-    assertThat(AnnotationRuleParser.guessType(boolean.class)).isEqualTo(PropertyType.BOOLEAN);
-    assertThat(AnnotationRuleParser.guessType(String.class)).isEqualTo(PropertyType.STRING);
-    assertThat(AnnotationRuleParser.guessType(Object.class)).isEqualTo(PropertyType.STRING);
+    Assertions.assertThat(AnnotationRuleParser.guessType(Integer.class)).isEqualTo(PropertyType.INTEGER);
```
Comment (jblievremont) [inline sonar-plugin-api/src/test/java/org/sonar/api/rules/AnnotationRuleParserTest.java]: ``Assertions.assertThat`` is already a static import, I don't get the point of these changes?

**Comment by jblievremont** [comment]
Very little feedback (and a few SQ issues ;) ), then LGTM

File: sonar-plugin-api/src/test/java/org/sonar/api/rules/AnnotationRuleParserTest.java (hunk 1 above)
Comment (simonbrandhof) [inline sonar-plugin-api/src/test/java/org/sonar/api/rules/AnnotationRuleParserTest.java]: I don't know where this comes from. Reverting.

**Comment by SonarTech** [comment]
SonarQube analysis reported 42 new issues:
* 8 critical
* 8 major
* 2 minor
* 24 info

Details:
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/batch/AbstractSourceImporter.java#L37)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/batch/ResourceCreationLock.java#L33)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/batch/ResourceFilter.java#L38)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/batch/SquidUtils.java#L47)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/batch/SquidUtils.java#L39)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/batch/SquidUtils.java#L29)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/batch/SquidUtils.java#L59)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [org.sonar.api.charts.AbstractChart.COLORS should be package protected](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/charts/AbstractChart.java#L44)[](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3AMS_PKGPROTECT)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/charts/AbstractChart.java#L39)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Make this array "private".](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/charts/AbstractChart.java#L44)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1873)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/charts/Chart.java#L34)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Provide multiple methods instead of using "decode" to determine which action to take.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/charts/ChartParameters.java#L97)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2301)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/charts/ChartParameters.java#L39)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Introduce a new variable instead of reusing the parameter "val".](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/charts/ChartParameters.java#L179)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1226)
* [Use an "instanceof" comparison instead.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/checks/AnnotationCheckFactory.java#L130)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1872)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/checks/AnnotationCheckFactory.java#L44)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Use an "instanceof" comparison instead.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/checks/AnnotationCheckFactory.java#L124)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1872)
* [Use an "instanceof" comparison instead.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/checks/AnnotationCheckFactory.java#L115)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1872)
* [Use an "instanceof" comparison instead.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/checks/AnnotationCheckFactory.java#L118)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1872)
* [The Cyclomatic Complexity of this method "configureField" is 15 which is greater than 10 authorized.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/checks/AnnotationCheckFactory.java#L108)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
* [Use an "instanceof" comparison instead.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/checks/AnnotationCheckFactory.java#L127)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1872)
* [Use an "instanceof" comparison instead.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/checks/AnnotationCheckFactory.java#L121)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1872)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/checks/CheckFactory.java#L34)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/checks/NoSonarFilter.java#L37)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [org.sonar.api.resources.Java.SUFFIXES should be package protected](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/resources/Java.java#L56)[](http://nemo.sonarqube.org/coding_rules#rule_key=findbugs%3AMS_PKGPROTECT)
* [Make this array "private".](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/resources/Java.java#L56)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1873)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/resources/Java.java#L34)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/resources/JavaFile.java#L35)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/resources/JavaPackage.java#L29)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/resources/Method.java#L26)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Override this superclass' "equals" method.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/resources/Method.java#L26)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2160)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/resources/ProjectUtils.java#L32)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/resources/ProjectUtils.java#L43)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Method 'Rule.setCardinality(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/rules/AnnotationRuleParser.java#L74)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/rules/AnnotationRuleParser.java#L46)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Provide multiple methods instead of using "useKeyIfEmpty" to determine which action to take.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/rules/RuleRepository.java#L60)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2301)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/rules/RuleRepository.java#L37)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Method 'Rule.setCardinality(...)' is deprecated.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/rules/XMLRuleParser.java#L153)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/rules/XMLRuleParser.java#L56)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [The Cyclomatic Complexity of this method "processRule" is 15 which is greater than 10 authorized.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/rules/XMLRuleParser.java#L118)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/web/AbstractDashboardWidget.java#L27)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)
* [Do not forget to remove this deprecated code someday.](https://github.com/SonarSource/sonarqube/blob/804c3fffc3ba2f9fdecca922bfd386c0d77cb975/sonar-plugin-api/src/main/java/org/sonar/api/web/Section.java#L27)[](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1133)

**Pull request closed without merging**